### PR TITLE
[+] Get last transaction hash

### PR DESCRIPTION
**Pull request by razvantomegea**

### CODE REVIEW

**Review by MiroMargineanu — APPROVED** [review]

File: src/hooks/transactions/useGetLastTransactionHash.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { useGetPendingTransactions } from 'hooks/transactions/useGetPendingTransactions';
+
+export const useGetLastTransactionHash = () => {
```
Comment (arhtudormorar) [inline src/hooks/transactions/useGetLastTransactionHash.ts:3]: `export const **useGetLatestTransactionHash** = (): string => {
`

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+- [Added `getLastTransaction` hook](https://github.com/multiversx/mx-sdk-dapp/pull/666)
```
Comment (arhtudormorar) [inline CHANGELOG.md:9]: getLastTransactionHash

File: src/hooks/transactions/useGetLastTransactionHash.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  const { hasPendingTransactions, pendingTransactions } =
+    useGetPendingTransactions();
```
Comment (arhtudormorar) [inline src/hooks/transactions/useGetLastTransactionHash.ts:5]: scenario:
1. Make a transaction
2. Sign transaction
3. getLatestTransactionHash --> OK
4. Transaction gets resolved
5. getLatestTransactionHash --> empty

I think getLatestTransactionHash should return a hash if one or more transactions have been processed

File: src/hooks/transactions/useGetLastTransactionHash.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  const sortedSessionIdsDesc = Object.keys(pendingTransactions)
+    .sort()
+    .reverse();
+
+  for (const sessionId of sortedSessionIdsDesc) {
+    const batch = pendingTransactions[sessionId];
+    const batchWithPendingTx = batch.transactions?.filter(
+      (tx) => !tx.status || tx.status === TransactionServerStatusesEnum.pending
```
Comment (arhtudormorar) [inline src/hooks/transactions/useGetLastTransactionHash.ts:19]: is `!tx.status` needed?

File: src/hooks/transactions/useGetLastTransactionHash.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { useGetPendingTransactions } from 'hooks/transactions/useGetPendingTransactions';
+import { TransactionServerStatusesEnum } from '../../types';
+
+export const useGetLastTransactionHash = (): string => {
```
Comment (arhtudormorar) [inline src/hooks/transactions/useGetLastTransactionHash.ts:4]: useGetLastPendingTransactionHash

File: src/hooks/transactions/useGetLastTransactionHash.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    .sort()
+    .reverse();
+
+  for (const sessionId of sortedSessionIdsDesc) {
+    const batch = pendingTransactions[sessionId];
+    const batchWithPendingTx = batch.transactions?.filter(
+      (tx) => !tx.status || tx.status === TransactionServerStatusesEnum.pending
+    );
+
+    if (batchWithPendingTx?.length) {
```
Comment (arhtudormorar) [inline src/hooks/transactions/useGetLastTransactionHash.ts:22]: batchWithPendingTx?.length > 0

File: CHANGELOG.md [hunk 7]
```diff
@@ -6,6 +6,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+- [Added `useGetLastTransactionHash` hook](https://github.com/multiversx/mx-sdk-dapp/pull/666)
```
Comment (arhtudormorar) [inline CHANGELOG.md:9]: useGetLastPendingTransactionHash

File: src/hooks/transactions/useGetPendingTransactions.ts [hunk 8]
```diff
@@ -2,18 +2,19 @@ import { useSelector } from 'reduxStore/DappProviderContext';
 import { pendingSignedTransactionsSelector } from 'reduxStore/selectors';
 import { SignedTransactionsType, SignedTransactionsBodyType } from 'types';
 
-export interface UseGetPendingTransactionsReturnType {
+export interface GetPendingTransactionsReturnType {
```
Comment (arhtudormorar) [inline src/hooks/transactions/useGetPendingTransactions.ts:5]: No renaming - might break stuff

**Review by arhtudormorar — APPROVED** [review]

**Pull request merged**